R/data.table.R

```diff
@@ -1340,7 +1340,9 @@ replace_dot_alias = function(e) {
     }
 
     if (is.data.table(jval)) {
-      setattr(jval, 'class', class(x)) # fix for #64
+      # should set the parent class only when jval is a plain data.table #4324
+      if (identical(class(jval), c('data.table', 'data.frame'))) 
+        setattr(jval, 'class', class(x)) # fix for #64
       if (haskey(x) && all(key(x) %chin% names(jval)) && is.sorted(jval, by=key(x)))
```

Comment:
There was a merge conflict here on the suppressWarnings() around `is.sorted()`. But it was clear that #4508 came later which removed the suppressWarnings().  This comment just serves for audit in case I got that wrong.